### ISSUE TITLE
Blank tabs

### ISSUE THREAD

**Issue by zalloy** [Nicotine-Plus/nicotine-plus#207]
I've been noticing an issue with Nicotine that's been occurring for some time now. I first noticed it just before I upgraded from Eoan to Focal, on both the stable and unstable branches. When I start Nicotine, all the downloads I had in progress disappeared and that tab is now blank. Additionally, the chat rooms tab is blank, nothing shows up when I try to search, and I'm assuming no one is downloading from me. My ports are open, as per the built-in checker, and I've double-checked that I've forwarded the ports in my router, disabled Xfinity (Comcast)'s "Advanced Security", and enabled the ports in UFW. 

I'm currently running Kubuntu Focal (20.04 LTS) with dark themes throughout (Adapta dark,Breeze dark,kvantum dark), and using the unstable branch for Nicotine (as I know there are lots of updates in progress for the port to Python 3). I tried changing my GTK themes to a light one just to see if it might be the dark theme causing problems, but there was no difference. I've also tried starting Nicotine from the terminal so I could see if there were any errors, but no errors are showing up. 

This is preventing me using Nicotine/Soulseek at all. I even tried installing SoulseekQT and had the same issue. Anybody have any ideas what might be causing this, and how I can fix it? Thanks in advance.

**Comment by mathiascode** [comment]
Anything of interest in the log view, if you enable debug info? (View -> Show debug info, check all the options)

**Comment by zalloy** [comment]
Nope, not seeing anything out of the ordinary. I'll attach a [screenshot](https://www.dropbox.com/s/jr4lv6qd4ufkvsc/Selection_524.png?dl=0). Not sure if there's a log file somewhere that might have more information.

**Comment by mathiascode** [comment]
Weird, that's what the program looks like when no connection has been established yet. Could server.slsknet.org:2242 be blocked somehow?

**Comment by zalloy** [comment]
Yeah, I thought it was weird too. I've got ports 2234:2239, 2240, and 2242 open for Soulseek/Nicotine, I've got the same ports forwarded on my router to this machine, and I also have server.slsknet.org whitelisted in my hosts file. Yet, it keeps on coming up like this, and if I click on the File menu, it's giving me the option to connect, as if I wasn't already connected to the server, but if I click that, nothing happens.

**Comment by mathiascode** [comment]
If you run `curl -I http://server.slsknet.org:2242`, do you get a `curl: (52) Empty reply from server` response?

**Comment by zalloy** [comment]
I tried running `curl -I http://server.slsknet.org:2242` and got no output, so then I tried pinging that server and got the same thing. This led me to believe that there was something else amiss, so I started checking things more thoroughly. 

In the end, I found out that a filter that I had set up in Fail2Ban was causing Nicotine/Soulseek to fail to connect. Removed that filter, and now I'm back in business. Going to go ahead and close this issue.
